extensions/wikia/UserProfilePageV3/tests/UserTagsStrategyTest.php

```diff
@@ -0,0 +1,190 @@
+			->method( 'isBlocked' )
+			->willReturn( $isBlocked );
+		$userMock->expects( $this->any() )
+			->method( 'getId' )
+			->willReturn( 1 );
+
+		$this->mockStaticMethod( WikiFactory::class, 'getWikiById', (object)[ 'city_founding_user' => (int) $isFounder ] );
+
+		$strategy = new UserTagsStrategy( $userMock );
+		$this->assertEquals( $expectedTags, $strategy->getUserTags() );
+	}
+
+	public function getUserTagsDataProvider() {
+		return [
+			'normal user' => [ [], [], false, false, false,
+				[]
+			],
+			'blocked user' => [ [], [], true, false, false,
+				[ wfMessage( 'user-identity-box-blocked' )->escaped() ]
+			],
+			'blocked user with other rights' => [ [ 'helper' ], [ 'sysop' ], true, true, true,
+				[ wfMessage( 'user-identity-box-blocked' )->escaped() ]
+			],
+			'blocked staff member' => [ [ 'staff' ], [], true, false, false,
+				[ wfMessage( 'user-identity-box-group-staff' )->escaped() ]
+			],
+			'blocked staff member with other rights' => [ [ 'staff' ], [ 'sysop' ], true, false, false,
+				[
+					wfMessage( 'user-identity-box-group-staff' )->escaped(),
+					wfMessage( 'user-identity-box-group-sysop' )->escaped()
+				]
+			],
+			'staff founder' => [ [ 'staff' ], [ 'bureaucrat', 'sysop' ], false, false, true,
+				[
+					wfMessage( 'user-identity-box-group-staff' )->escaped(),
+					wfMessage( 'user-identity-box-founder' )->escaped()
+				]
+			],
+			'founder with other global rights' => [ [ 'helper', 'vstf' ], [ 'bureaucrat', 'sysop' ], false, false, true,
+				[
+					wfMessage( 'user-identity-box-group-helper' )->escaped(),
+					wfMessage( 'user-identity-box-founder' )->escaped()
+				]
+			],
+			'staff and local admin' => [ [ 'staff' ], [ 'sysop' ], false, false, false,
+				[
+					wfMessage( 'user-identity-box-group-staff' )->escaped(),
+					wfMessage( 'user-identity-box-group-sysop' )->escaped()
+				]
+			],
+			'multiple global rights and local admin' => [ [ 'vstf', 'council' ], [ 'sysop' ], false, false, false,
+				[
+					wfMessage( 'user-identity-box-group-vstf' )->escaped(),
+					wfMessage( 'user-identity-box-group-sysop' )->escaped()
+				]
+			],
+			'local rights only' => [ [], [ 'bureaucrat', 'threadmoderator', 'sysop' ], false, false, false,
+				[
+					wfMessage( 'user-identity-box-group-bureaucrat' )->escaped(),
+					wfMessage( 'user-identity-box-group-sysop' )->escaped()
+				]
+			],
+			'global rights only' => [ [ 'staff', 'vstf' ], [], false, false, false,
+				[
+					wfMessage( 'user-identity-box-group-staff' )->escaped(),
+					wfMessage( 'user-identity-box-group-vstf' )->escaped()
+				]
+			],
+			'founder with no global and multiple local rights' => [ [], [ 'threadmoderator', 'sysop' ], false, false, true,
+				[
+					wfMessage( 'user-identity-box-group-sysop' )->escaped(),
+					wfMessage( 'user-identity-box-founder' )->escaped()
+				]
+			],
+			'banned from chat' => [ [], [], false, true, false,
+				[ wfMessage( 'user-identity-box-banned-from-chat' )->escaped() ]
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "user-identity-box-banned-from-chat" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)